rotkehlchen/globaldb/upgrades/v7_v8.py

```diff
@@ -66,11 +66,54 @@ def fix_detected_spam_tokens(write_cursor: 'DBCursor') -> None:
     )
 
 
+@enter_exit_debug_log()
+def set_unique_asset_collections(cursor: 'DBCursor') -> None:
+    """It does the following:
+    - Create a new asset_collections table with UNIQUE constraint and a temporary name.
+    - Copy all the asset collections to the new table, ignoring the duplicated ones.
+    - Delete the old table.
+    - Rename the new table to the old one.
+    """
+    v7_multiasset_mappings = cursor.execute(
+        'SELECT rowid, collection_id, asset FROM multiasset_mappings;',
+    ).fetchall()
+    cursor.execute("""
+        CREATE TABLE IF NOT EXISTS asset_collections_temp(
+            id INTEGER PRIMARY KEY,
+            name TEXT NOT NULL,
+            symbol TEXT NOT NULL,
+            UNIQUE (name, symbol)
+        );
+    """)
+
+    all_entries = cursor.execute('SELECT id, name, symbol FROM asset_collections;').fetchall()
+    unique_entries = set()
+    duplicated_entries = 0
+    for collection_id, name, symbol in all_entries:
+        if (name, symbol) not in unique_entries:
+            cursor.execute(
+                'INSERT INTO asset_collections_temp(id, name, symbol) VALUES (?, ?, ?);',
+                (collection_id - duplicated_entries, name, symbol),
```

Comment:
this assumes consecutive indices... is that the case?

Reply:
no it's considering that they are not consecutive. For example, in the current db the the collection id 249 is missing, so that will be kept as it is. But, if we find any duplicate entry then the next one will be placed at its place.